toolbox/db/db_get.m

```diff
@@ -218,32 +218,58 @@
         else
             varargout{1} = sAnatFiles;
         end
-
-    % Usage: sFiles = db_get('FunctionalFile', FileType, FileIDs)
-    % Usage: sFiles = db_get('FunctionalFile', FileType, FileNames)
+        
+%% ==== FUNCTIONAL FILE ====
+    % Usage: [sFiles, sItems] = db_get('FunctionalFile', FileIDs,   Fields)
+    % Usage: [sFiles, sItems] = db_get('FunctionalFile', FileNames, Fields)
+    % Usage: [sFiles, sItems] = db_get('FunctionalFile', CondQuery, Fields)
     case 'FunctionalFile'
-        type = args{1};
-        iFiles = args{2};
+        % Parse inputs
+        iFiles = args{1};
+        fields = '*';                              
+        templateStruct = db_template('FunctionalFile');
+
         if ischar(iFiles)
             iFiles = {iFiles};
+        elseif isstruct(iFiles)
+            condQuery = CondQuery;           
         end
-        nFiles = length(iFiles);
-        sFiles = repmat(db_template('FunctionalFile'), 1, nFiles);
-        sItems = repmat(db_template(type), 1, nFiles);
 
+        if length(args) > 1
+            fields = args{2};
+            if ischar(fields)
+                fields = {fields};
+            end
+            for i = 1 : length(fields)
+                resultStruct.(fields{i}) = templateStruct.(fields{i});
+            end
+        else
+            resultStruct = templateStruct;
+        end
+              
+        nFiles = length(iFiles);
+        sFiles = repmat(resultStruct, 1, nFiles);
+        sItems= [];
+        
         for i = 1:nFiles
             if iscell(iFiles)
```

Comment:
Now that iFiles is an optional argument, this condition does not handle all cases (when iFiles won't be set), e.g.
> db_get('FunctionalFile', sqlConn, struct('Type', 'folder', 'Study', 7))

Reply:
The update in daed99c checks for this case for struct query as input